### ISSUE TITLE
Proposal: Allow setters to set value without a dummy variable

### ISSUE THREAD

**Issue by KakCAT** [dotnet/csharplang#2010]
Hello,

```
public MyEnumType data 
{
  get { return data; }
  set { data = value; }
}
```
causes StackOverflowException. I understand why, and I have to add a dummy variable (i.e. _data) and work with that dummy variable. (and I'm aware that for this example I just could use "get; set;" but this is obviously a minimal example)

Is there / Would it be possible to have a mecanism to avoid having to declare the same variable twice and yet be able to assign it without StackOverflowEx ? (it'd be probably just a case of syntactic sugar, as once you declare "_data" and use it in "data", the memory space of "data" disappears)

something on the line of 
```
set { MyEnumType.data = value; }

or

set { settedFieldValue = value; }

or

set { data.value = value; } 
```
( I'm not good with naming :) )

Thanks, and sorry if this request is duplicated.

**Comment by DavidArno** [comment]
Please see #140. Not a direct duplicate, but it contains a lot of discussion on this topic.

**Comment by KakCAT** [comment]
> 
> 
> Please see #140. Not a direct duplicate, but it contains a lot of discussion on this topic.

Thanks for the answer. I'm closing it because despite not being a direct duplicate of #140, reading it I've found reading that #1805 is a duplicate (and was closed and referred to #140)

**Comment by jnm2** [comment]
Considered in https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-01.md
